apps/rush-lib/src/logic/base/BaseInstallManager.ts

```diff
@@ -150,12 +155,43 @@ export abstract class BaseInstallManager {
   }
 
   public async doInstall(): Promise<void> {
+    const isFilteredInstall: boolean = this.options.toFlags.length > 0;
+
+    // Prevent filtered installs when workspaces is disabled
+    if (
+      isFilteredInstall &&
+      !(this.rushConfiguration.pnpmOptions && this.rushConfiguration.pnpmOptions.useWorkspaces)
+    ) {
+      console.log();
+      console.log(
+        colors.red(
+          'The "--to" argument can only be used when running in a workspace environment. Run the ' +
+            'command again without specifying this argument.'
+        )
+      );
+      throw new AlreadyReportedError();
+    }
+
+    // Prevent update when using a filter, as modifications to the shrinkwrap shouldn't be saved
+    if (this.options.allowShrinkwrapUpdates && isFilteredInstall) {
+      console.log();
+      console.log(
+        colors.red(
+          'The "--to" argument cannot be used when running "rush update". Run the command again ' +
+            'without specifying this argument.'
+        )
+      );
+      throw new AlreadyReportedError();
+    }
+
     const { shrinkwrapIsUpToDate, variantIsUpToDate } = await this.prepareAsync();
 
     // This marker file indicates that the last "rush install" completed successfully.
-    // If "--purge" was specified, or if the last install was interrupted, then we will need to
-    // perform a clean install.  Otherwise, we can do an incremental install.
-    const cleanInstall: boolean = !this._commonTempInstallFlag.checkValidAndReportStoreIssues();
+    // Always perform a clean install if filter flags were provided. Additionally, if
```

Comment:
Correct, we do purge out the common/temp folder for every filtered install. I had explored doing some additional work around possibly adding the args/filtered projects to `last-install.flag`, but basically reached the conclusion that since a filtered install is basically a purposely incomplete install, that this was the best approach to ensure that devs don't get into some weird state.